File: tradebusiness-admin/src/api/leadGeneration.ts
/**
 * AI 获客 API
 */
import request from '@/utils/request'

export interface LeadSearchTask {
  id: string
  user_id: string
  name: string
  target_country?: string
  target_industry?: string
  product_keywords: string[]
  customer_profile?: string
  exclude_keywords: string[]
  website_inputs: string[]
  status: 'pending' | 'running' | 'completed' | 'failed' | 'cancelled'
  total_found: number
  error_message?: string
  created_at: string
  updated_at: string
}

export interface LeadSearchTaskCreate {
  name: string
  target_country?: string
  target_industry?: string
  product_keywords: string[]
  customer_profile?: string
  exclude_keywords: string[]
  website_inputs: string[]
}

export interface Lead {
  id: string
  task_id?: string
  user_id: string
  company_name: string
  website?: string
  country?: string
  industry?: string
  description?: string
  source?: string
  match_score?: number
  status: 'new' | 'researched' | 'copy_generated' | 'contacted' | 'replied' | 'invalid'
  do_not_contact: boolean
  do_not_contact_reason?: string
  notes?: string
  created_at: string
  updated_at: string
}

export interface CompanyResearchReport {
  id: string
  lead_id: string
  summary: string
  business_model?: string
  products?: string
  target_markets?: string
  buying_signals?: string
  pain_points?: string
  recommended_angle?: string
  raw_sources?: Record<string, unknown>
  created_at: string
  updated_at: string
}

export interface SalesCopy {
  id: string
  lead_id: string
  channel: 'email' | 'whatsapp' | 'linkedin'
  language: string
  subject?: string
  content: string
  tone?: string
  status: string
  created_at: string
  updated_at: string
}

export function getLeadTasks(params?: { skip?: number; limit?: number; status?: string }) {
  return request<{ items: LeadSearchTask[]; total: number }>({
    url: '/api/v1/lead-generation/tasks',
    method: 'get',
    params,
  })
}

export function createLeadTask(data: LeadSearchTaskCreate) {
  return request<LeadSearchTask>({
    url: '/api/v1/lead-generation/tasks',
    method: 'post',
    data,
  })
}

export function runLeadTask(id: string) {
  return request<LeadSearchTask>({
    url: `/api/v1/lead-generation/tasks/${id}/run`,
    method: 'post',
  })
}

export function cancelLeadTask(id: string) {
  return request<LeadSearchTask>({
    url: `/api/v1/lead-generation/tasks/${id}/cancel`,
    method: 'post',
  })
}

export function getLeadTask(id: string) {
  return request<LeadSearchTask>({
    url: `/api/v1/lead-generation/tasks/${id}`,
    method: 'get',
  })
}

export function getLeads(params?: {
  skip?: number
  limit?: number
  task_id?: string
  status?: string
  search?: string
}) {
  return request<{ items: Lead[]; total: number }>({
    url: '/api/v1/lead-generation/leads',
    method: 'get',
    params,
  })
}

export function createLead(data: Partial<Lead>) {
  return request<Lead>({
    url: '/api/v1/lead-generation/leads',
    method: 'post',
    data,
  })
}

export function generateCompanyResearch(leadId: string) {
  return request<CompanyResearchReport>({
    url: `/api/v1/lead-generation/leads/${leadId}/research`,
    method: 'post',
  })
}

export function getCompanyResearch(leadId: string) {
  return request<CompanyResearchReport | null>({
    url: `/api/v1/lead-generation/leads/${leadId}/research`,
    method: 'get',
  })
}

export function generateSalesCopy(
  leadId: string,
  data: { channel: 'email' | 'whatsapp' | 'linkedin'; language: string; tone?: string; agent_id?: string }
) {
  return request<SalesCopy>({
    url: `/api/v1/lead-generation/leads/${leadId}/sales-copy`,
    method: 'post',
    data,
  })
}

export function getSalesCopies(leadId: string) {
  return request<SalesCopy[]>({
    url: `/api/v1/lead-generation/leads/${leadId}/sales-copies`,
    method: 'get',
  })
}

export function importLeadsCsv(file: File, taskId?: string) {
  const formData = new FormData()
  formData.append('file', file)
  return request<{ created: number }>({
    url: '/api/v1/lead-generation/leads/import',
    method: 'post',
    params: taskId ? { task_id: taskId } : undefined,
    data: formData,
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}
export interface LeadContact { id: string; lead_id: string; name?: string; title?: string; email?: string; phone?: string; linkedin_url?: string; whatsapp?: string; source?: string; is_verified: boolean; created_at: string; updated_at: string }
export function getLeadDetail(id: string) { return request<Lead>({ url: `/api/v1/lead-generation/leads/${id}`, method: 'get' }) }
export function updateLead(id: string, data: { do_not_contact: boolean; do_not_contact_reason?: string }) { return request<Lead>({ url: `/api/v1/lead-generation/leads/${id}`, method: 'patch', data }) }
export function getLeadContacts(id: string) { return request<LeadContact[]>({ url: `/api/v1/lead-generation/leads/${id}/contacts`, method: 'get' }) }
export function createLeadContact(id: string, data: Partial<LeadContact>) { return request<LeadContact>({ url: `/api/v1/lead-generation/leads/${id}/contacts`, method: 'post', data: { ...data, lead_id: id } }) }

export function sendLeadEmail(id: string, data: { to_email: string; to_name?: string; subject: string; body: string }) { return request<{ id: string; status: string; lead_id: string }>({ url: `/api/v1/lead-generation/leads/${id}/send-email`, method: 'post', data }) }
export interface FollowupTask { id: string; lead_id: string; channel: string; subject?: string; content: string; due_at: string; status: string }
export function getFollowups(channel?: string) { return request<FollowupTask[]>({ url: '/api/v1/lead-generation/followups', method: 'get', params: channel ? { channel } : undefined }) }
export function createFollowup(data: Partial<FollowupTask>) { return request<FollowupTask>({ url: '/api/v1/lead-generation/followups', method: 'post', data }) }
export function updateFollowup(id: string, status: string) { return request<FollowupTask>({ url: `/api/v1/lead-generation/followups/${id}`, method: 'patch', params: { status } }) }
export function retryFollowup(id: string) { return request<FollowupTask>({ url: `/api/v1/lead-generation/followups/${id}/retry`, method: 'post' }) }

export function getLeadGenerationStats() { return request<{ leads: number; sales_copies: number; followups: number; completed_followups: number; channels: Record<string, number>; countries: Record<string, number>; industries: Record<string, number>; statuses: Record<string, number> }>({ url: '/api/v1/lead-generation/stats', method: 'get' }) }

File: tradebusiness-backend/app/models/lead_generation.py
"""
AI 获客模型
定义获客任务、潜客、联系人、企业背调和开发信的数据结构。
"""

from datetime import datetime
from enum import Enum
from typing import Optional

from sqlalchemy import JSON, DateTime, Float, ForeignKey, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, TimestampMixin, UUIDMixin


class LeadTaskStatus(str, Enum):
    """获客任务状态。"""

    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"
    CANCELLED = "cancelled"


class LeadStatus(str, Enum):
    """潜客状态。"""

    NEW = "new"
    RESEARCHED = "researched"
    COPY_GENERATED = "copy_generated"
    CONTACTED = "contacted"
    REPLIED = "replied"
    INVALID = "invalid"


class SalesCopyChannel(str, Enum):
    """开发话术渠道。"""

    EMAIL = "email"
    WHATSAPP = "whatsapp"
    LINKEDIN = "linkedin"


class LeadSearchTask(Base, UUIDMixin, TimestampMixin):
    """AI 获客搜索任务。"""

    user_id: Mapped[str] = mapped_column(
        String(36),
        ForeignKey("users.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    name: Mapped[str] = mapped_column(String(200), nullable=False, index=True)
    target_country: Mapped[Optional[str]] = mapped_column(String(100), nullable=True, index=True)
    target_industry: Mapped[Optional[str]] = mapped_column(String(120), nullable=True, index=True)
    product_keywords: Mapped[Optional[dict]] = mapped_column(JSON, nullable=True)
    customer_profile: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    exclude_keywords: Mapped[Optional[dict]] = mapped_column(JSON, nullable=True)
    website_inputs: Mapped[Optional[dict]] = mapped_column(JSON, nullable=True)
    status: Mapped[str] = mapped_column(
        String(20),
        default=LeadTaskStatus.PENDING.value,
        nullable=False,
        index=True,
    )
    total_found: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    error_message: Mapped[Optional[str]] = mapped_column(Text, nullable=True)


class Lead(Base, UUIDMixin, TimestampMixin):
    """AI 获客潜客公司。"""

    task_id: Mapped[Optional[str]] = mapped_column(
        String(36),
        ForeignKey("lead_search_tasks.id", ondelete="SET NULL"),
        nullable=True,
        index=True,
    )
    user_id: Mapped[str] = mapped_column(
        String(36),
        ForeignKey("users.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    company_name: Mapped[str] = mapped_column(String(200), nullable=False, index=True)
    website: Mapped[Optional[str]] = mapped_column(String(500), nullable=True, index=True)
    country: Mapped[Optional[str]] = mapped_column(String(100), nullable=True, index=True)
    industry: Mapped[Optional[str]] = mapped_column(String(120), nullable=True, index=True)
    description: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    source: Mapped[Optional[str]] = mapped_column(String(100), nullable=True)
    match_score: Mapped[Optional[float]] = mapped_column(Float, nullable=True)
    status: Mapped[str] = mapped_column(
        String(30),
        default=LeadStatus.NEW.value,
        nullable=False,
        index=True,
    )
    do_not_contact: Mapped[bool] = mapped_column(default=False, nullable=False, index=True)
    do_not_contact_reason: Mapped[Optional[str]] = mapped_column(String(300), nullable=True)
    notes: Mapped[Optional[str]] = mapped_column(Text, nullable=True)


class LeadContact(Base, UUIDMixin, TimestampMixin):
    """潜客联系人。"""

    lead_id: Mapped[str] = mapped_column(
        String(36),
        ForeignKey("leads.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    name: Mapped[Optional[str]] = mapped_column(String(120), nullable=True)
    title: Mapped[Optional[str]] = mapped_column(String(120), nullable=True)
    email: Mapped[Optional[str]] = mapped_column(String(120), nullable=True, index=True)
    phone: Mapped[Optional[str]] = mapped_column(String(60), nullable=True)
    linkedin_url: Mapped[Optional[str]] = mapped_column(String(500), nullable=True)
    whatsapp: Mapped[Optional[str]] = mapped_column(String(60), nullable=True)
    source: Mapped[Optional[str]] = mapped_column(String(100), nullable=True)
    is_verified: Mapped[bool] = mapped_column(default=False, nullable=False)


class CompanyResearchReport(Base, UUIDMixin, TimestampMixin):
    """企业背调报告。"""

    lead_id: Mapped[str] = mapped_column(
        String(36),
        ForeignKey("leads.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    summary: Mapped[str] = mapped_column(Text, nullable=False)
    business_model: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    products: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    target_markets: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    buying_signals: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    pain_points: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    recommended_angle: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    raw_sources: Mapped[Optional[dict]] = mapped_column(JSON, nullable=True)


class SalesCopy(Base, UUIDMixin, TimestampMixin):
    """AI 生成的开发信和渠道话术。"""

    __tablename__ = "sales_copies"

    lead_id: Mapped[str] = mapped_column(
        String(36),
        ForeignKey("leads.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    channel: Mapped[str] = mapped_column(String(30), nullable=False, index=True)
    language: Mapped[str] = mapped_column(String(20), default="en", nullable=False)
    subject: Mapped[Optional[str]] = mapped_column(String(300), nullable=True)
    content: Mapped[str] = mapped_column(Text, nullable=False)
    tone: Mapped[Optional[str]] = mapped_column(String(50), nullable=True)
    status: Mapped[str] = mapped_column(String(30), default="draft", nullable=False)


class FollowupTask(Base, UUIDMixin, TimestampMixin):
    """潜客后续跟进任务。"""

    lead_id: Mapped[str] = mapped_column(
        String(36), ForeignKey("leads.id", ondelete="CASCADE"), index=True
    )
    user_id: Mapped[str] = mapped_column(
        String(36), ForeignKey("users.id", ondelete="CASCADE"), index=True
    )
    channel: Mapped[str] = mapped_column(String(30), default="email", nullable=False)
    subject: Mapped[Optional[str]] = mapped_column(String(300), nullable=True)
    content: Mapped[str] = mapped_column(Text, nullable=False)
    due_at: Mapped[datetime] = mapped_column(DateTime, nullable=False, index=True)
    status: Mapped[str] = mapped_column(String(30), default="pending", nullable=False, index=True)
